### PR TITLE
Split tests and allocation into anon and subs

### PR DESCRIPTION
**Pull request by tom-parker**
cc @commuterjoy 

Flags will need to pass `cohort: "subscriber/anonymous/all"` to be included in that group. Any AB flags with a missing `cohort` attribute will be considered for everyone.

Users are only allocated the subscriber tests when they have a valid session.

### CODE REVIEW

**Comment by commuterjoy** [comment]
+1

File: server/middleware/set-ab-tests.js [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
 	// Note: res.locals.flagsArray is provided by next-express.
-	var flagsWithABTests = res.locals.flagsArray.filter(function (flag) {
+	const flagsWithABTests = res.locals.flagsArray.filter(function (flag) {
 		return flag.abTestState === true;
 	});
 
+	const anonymousTests = flagsWithABTests.filter(function (flag) {
+		// Group flags for anon users. Flags without a cohort are assumed to be
+		// for all users
+		return flag.cohort === 'anonymous' || flag.cohort === 'all' || !flag.cohort;
```
Comment (commuterjoy) [inline server/middleware/set-ab-tests.js]: i suppose at some point we want to mandate a cohort in the flags model rather than leaving it undefined.

**Comment by commuterjoy** [comment]
@tom-parker how do you feel about 4ed93dc ?

**Comment by tom-parker** [comment]
@commuterjoy yep, looks good. Is it just a style thing or is there some forward thinking plan?

**Pull request merged**